### ISSUE TITLE
Gradient Explosion when Finetuning from SiT

### ISSUE THREAD

**Issue by zheyuanhu01** [zhuyu-cs/MeanFlow#22]
Dear Author,

Thank you again for your great implementation!

I was trying to reproduce your results in the "finetune_fm" branch. I follow the instruction of not using CFG at the beginning, but found the gradient norm is extremely large ~1e3. I have to reduce the learning rate from 1e-4 to 1e-6.

I wonder if you have a similar observation.

Best regards,
Zheyuan

**Comment by zhuyu-cs** [comment]
Use fp32 could solve this problem.
